server/room-battle.js

```diff
@@ -704,6 +705,11 @@ class RoomBattle extends RoomGames.RoomGame {
 		switch (lines[0]) {
 		case 'update':
 			for (const line of lines.slice(1)) {
+				const tokens = line.slice(1).split("|");
```

Comment:
This sounds like overkill for `line.startsWith('|turn|')`

Reply:
All the unnecessary tokenizing/parsing/allocation...

Reply:
Done here https://github.com/Zarel/Pokemon-Showdown/pull/5685/commits/7b2d824a11d1112fd352c1235ed048ddb826bd26.